analytics/transform.go

```diff
@@ -191,22 +192,116 @@ func (event ParsedEvent) GetValue(field string) (interface{}, error) {
 		return nil, errors.New(fmt.Sprintf("Key %s not a valid atomic field", field))
 	}
 	if event[index] == "" {
-		return nil, errors.New(fmt.Sprintf("Field %s is empty", field))
+		return nil, errors.New(EmptyFieldErr)
 	}
 	kvPairs, err := enrichedEventFieldTypes[index].ParseFunction(enrichedEventFieldTypes[index].Key, event[index])
 	if err != nil {
 		return nil, err
 	}
-	if field == "contexts" || field == "derived_contexts" || field == "unstruct_event" {
-		// TODO: DRY HERE TOO?
-		output := make(map[string]interface{})
+
+	return kvPairs, nil
+}
+
+// GetValue returns the value for a provided atomic field, without processing the rest of the event.
+// For unstruct_event, it returns a map of only the data for the unstruct event.
+func (event ParsedEvent) GetValue(field string) (interface{}, error) {
+	kvPairs, err := event.getParsedValue(field)
+	if err != nil {
+		return nil, err
+	}
+	return kvPairs[0].Value, nil
```

Comment:
There's a regression here which seems problematic to me. This piece of code was removed:

```
if field == "contexts" || field == "derived_contexts" || field == "unstruct_event" {
		// TODO: DRY HERE TOO?
		output := make(map[string]interface{})
		for _, pair := range kvPairs {
			output[pair.Key] = pair.Value
		}
		return output, nil
	}
```

In the case of contexts, this results in a) A different output data structure, and b) an incomplete result set, only the first context is returned.

In the case of unstruct event, it results in a different output data structure.

I'm not sure if that's an oversight or deliberate, and if deliberate what the line of reasoning is - but as I see it the advantage of introducing new methods for the functionality in this PR is to avoid a breaking change to the existing methods. I don't think we should change the behaviour of the existing method.

Reply:
To clarify, in case this is where it came from - when I suggested we could return just the first context we find as a first iteration, I meant in the GetContextValue function. :)

Reply:
As per our discussion, I will add this bit of code back in order to prevent any regression.